.github/workflows/test.yml

```diff
@@ -2,9 +2,8 @@ name: Tests
 
 on:
   pull_request:
-  push:
```

Comment:
If someone pushes commits to a new branch not mentioned in the list below, tests will not be executed. That does not look great.
Instead it's better to limit branch names which are triggering the workflow in case of creating new pull request, because most of them are created with target_branch=master/main.

Reply:
Could we remove the branch restriction instead and run tests on all pushes? Is anyone aware of a reason to keep it?

Reply:
The only reason of  using such a restriction I know is a bit unexpected Github Actions behavior. In case then you've pushed a commit into some branch and then created a pull request using this branch as a source, 2 copies of workflow will be started - one for `push` event and another one for `pull_request` event.

If someone is using Git Flow (way of naming and working with branches in large projects), it can be quite common situation. For example, you've created branch `feature/abc` and then created a pull request into `dev` or `release-X.Y` branch of the same repo. All the tests here will be run twice. But limiting a list of target branches of pull request could prevent this situation.

If pull requests are created mostly from forked repos instead of branches of the same repo, you'll never face with such an issue, so this restriction could be removed.

Reply:
A note on the suggested development flow (in case that helps decide things) here: fork&PR is the recommended way to make changes, even for maintainers. We encourage people to open a work-in-progress PR as soon as possible, even if they are just doing exploratory work. This means there should almost never be additional branches in the upstream repository and most contributors get the benefits of CI "for free".

We do not have `release-X.Y` branches as all changes target HEAD and each merge into the main branch is a "release". We never backport changes to earlier "releases".

However some of the maintainers sometimes create a branch (a la `betatim-patch-1` etc) when to fix a small typo or such. And then make a PR. In these cases we'd have the tests run twice.

Overall it seems there are reasons to keep what is in the config now and reasons to change. Going for the principle of "make the smallest number of changes possible": what workflow/positive feature that is currently not possible would be enabled by making this change?

Reply:
Well, for example I've found few bugs, created a fork and made several branches for each bug to then create separated pull requests.
But now pushing commits leads to running tests only if in case of master branch. So I'm not sure that my changes are not causing some issues until I made a pull request to the original repo.

Reply:
I'm strongly in favour of running CI tests on pushes to all branches in all forks unless there's a reason not to, because it means contributors can experiment in their own forks without opening PRs. This is particularly the case now that people opening their first PR need to be approved before CI tests are run (an unfortunate consequence of cryptominers).

Reply:
Any updates here?

Reply:
My preference is for
```yaml
on:
  pull_request:
  push:
  workflow_dispatch:
```
and not restrict this to particular branches. Same for `diff.yml`

Reply:
OK, fixed this.